src/EditorFeatures/CSharp/CommentBlockCompletion/CommentBlockCompletionCommandHandler.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editor.Host;
+using Microsoft.CodeAnalysis.Editor.Implementation.CommentBlockCompletion;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Operations;
+using Microsoft.VisualStudio.Utilities;
+using Microsoft.CodeAnalysis.Text.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.CommentBlockCompletion
+{
+    [ExportCommandHandler(PredefinedCommandHandlerNames.CommentBlockCompletion, ContentTypeNames.CSharpContentType)]
+    internal class CommentBlockCompletionCommandHandler : AbstractCommentBlockCompletionCommandHandler
+    {
+        [ImportingConstructor]
+        public CommentBlockCompletionCommandHandler(
+            ITextUndoHistoryRegistry undoHistoryRegistry,
+            IEditorOperationsFactoryService editorOperationsFactoryService) : base(undoHistoryRegistry, editorOperationsFactoryService)
+        {
+        }
+
+        protected override string GetExteriorTextForNextLine(SnapshotPoint caretPosition)
+        {
+            var snapshotLine = caretPosition.Snapshot.GetLineFromPosition(caretPosition);
+
+            var firstNonWhitespacePosition = snapshotLine.GetFirstNonWhitespacePosition() ?? -1;
+            if (firstNonWhitespacePosition == -1)
+            {
+                return null;
+            }
+
+            var isCurrentLineStartsWithCommentBlockStart = snapshotLine.StartsWith(firstNonWhitespacePosition, "/*", false);
```

Comment:
I'm also a non-native English speaker, too. My bad. Should I use `does`? Or just remove the prefix? Same question for the `IsCommentBlockEndsAfterCaret` method below.